firmware/user/modes/mode_flight.c

```diff
@@ -0,0 +1,1077 @@
+    plotText(0, 0, framesStr, TOM_THUMB, WHITE);
+    int ij = tflight->frames;
+    SetupMatrix();
+
+
+#ifndef EMU
+    PIN_FUNC_SELECT( PERIPHS_IO_MUX_U0TXD_U, 3); //Set to GPIO.  
+    GPIO_OUTPUT_SET(GPIO_ID_PIN(1), 0 );
+    OVERCLOCK_SECTION_ENABLE();
+#endif
+
+//#define ORIG_TEST
+#ifdef ORIG_TEST
+    int x = 0;
+    int y = -1;
+    tdRotateEA( ProjectionMatrix, -20, 0, 0 );
+    tdRotateEA( ModelviewMatrix, ij, 0, 0 );
+    ij = 0;    //Uncomment to prevent animation (for perf test)
+    for( x = -3; x < 4; x++ )
+    {
+        for( y = 0; y < 4; y++ )
+        {
+            //7 * 4 * 120 = 3360 lines per frame.
+            ModelviewMatrix[11] = 2400;
+            ModelviewMatrix[3] = 500*x-800;
+            ModelviewMatrix[7] = 500*y+500;
+            tdDrawModel( tflight->isosphere );
+        }
+    }
+#else
+    int x = 0;
+    int y = -1;
+    tdRotateEA( ProjectionMatrix, tflight->hpr[1], tflight->hpr[0], 0 );
+	tdTranslate( ModelviewMatrix, tflight->planeloc[0], tflight->planeloc[1], tflight->planeloc[2] );
+
+	int16_t BackupMatrix[16];
+    for( x = -6; x < 17; x++ )
+    {
+        for( y = -2; y < 10; y++ )
+        {
+            memcpy( BackupMatrix, ModelviewMatrix, sizeof( BackupMatrix ) );
+			tdTranslate( ModelviewMatrix, 
+				500*x-800,
+				140 + tdSIN( (x + y)*40 + ij*1 )>>2,
+  		        500*y+500 );
+			tdScale( ModelviewMatrix, 70, 70, 70 );
+            tdDrawModel( tflight->isosphere );
+            memcpy( ModelviewMatrix, BackupMatrix, sizeof( BackupMatrix ) );
+        }
+    }
+#endif
+
+#ifndef EMU
+    OVERCLOCK_SECTION_DISABLE();
+    GPIO_OUTPUT_SET(GPIO_ID_PIN(1), 1 ); 
+#endif
+}
+
+static void ICACHE_FLASH_ATTR flightGameUpdate( flight_t * tflight )
+{
+    uint8_t bs = tflight->buttonState;
+
+
+    if( bs & 1 ) tflight->hpr[0]++;
+    if( bs & 4 ) tflight->hpr[0]--;
+    if( bs & 2 ) tflight->hpr[1]++;
+    if( bs & 8 ) tflight->hpr[1]--;
+
+	//Left-right is [0].
+	//Up-down is [1].
+
+    tflight->planeloc[0] -= tdSIN( tflight->hpr[0] )>>6;
+    tflight->planeloc[2] -= tdCOS( tflight->hpr[0] )>>6;
+    tflight->planeloc[1] += tdSIN( tflight->hpr[1] )>>6;
+
+/*
+    tflight->planeloc[0] += tdSIN( tflight->hpr[0] )>>4;
+    tflight->planeloc[1] += tdCOS( tflight->hpr[0] )>>4;
+    tflight->planeloc[2] += tdCOS( tflight->hpr[1] )>>4;
+
+    int i;
+    int16_t xformlast[3];
+    int newseg = 1;
+    for( i = 0; ; i++ )
+    {
+        if( newseg )
+        {
+            if( vTransform( tflight, xformlast, pittsburg+i*3 ) == 0 )
+            {
+                break;
+            }
+            
+            newseg = 0;
+        }
+        else
+        {
+            int16_t xformednow[3];
+            if( vTransform( tflight, xformednow, pittsburg+i*3 ) == 0 ) { newseg = 1; continue; }
+            plotLine( xformlast[0], xformlast[1], xformednow[0], xformednow[1], WHITE );
+            ets_memcpy( xformlast, xformednow, sizeof( xformednow) );
+        }
+    }
+*/
+
+#if 0
```

Comment:
I think it's safe to junk all this. If you need to reference it, it'll be in the other file

Reply:
will do.